api/ocm/plugin/README.md

```diff
@@ -0,0 +1,151 @@
+# The OCM Library Plugin Concept
+
+The ocm library supports a plugin mechanism to provide further variants
+for OCM and library extension points without the need of extending and recompiling
+the OCM CLI (and other applications using this library).
+
+This plugin concept is not part of the OCM specification, because it is
+a feature of this library implementation.
+
+The following extension points are supported:
+
+- Access methods
+- Uploaders
+- Downloaders
+- Actions
+- Value sets (for example for routing slip entries)
+- Config types
+- Value Merge Handler (for example for label values in delta transports)
+- CLI Commands (for OCM CLI)
+- (transfer handlers)
+- (signing tools)
+- (input types for component version composition)
+
+## Plugin Technology
+
+A plugin is a simple executable, which might be written in any program language.
+The plugin has to provide a set of CLI commands for every extension point it provides new variations for.
+
+The data transfer between the library and the plugin is done via
+
+- command options (for inbound information)
+- standard input (for potentially large inbound content (streaming))
+- standard output (for structured outbound information)
+- standard output (for unstructured data (streaming)
+
+The commands and their interface are described in the [plugin reference](../../../docs/pluginreference/plugin.md).
+
+Every plugin must provide the [`info`](../../../docs/pluginreference/plugin_info.md) command. It has to provide information
+about the supported features as JSOn document on standard output.
+The structure of this document is described by the descriptor type in package [`api/ocm/plugin/descriptor`](descriptor/descriptor.go)
+
+Plugin are searched in a plugin folder (typically `.ocm/plugins`), This default location can be changed by the `plugincachedir` attribute.
+
+## Plugin-related CLI Commands
+
+The OCM CLI provides commands to
+
+- [install](../../../docs/reference/ocm_install_plugins.md)
+- [update](../../../docs/reference/ocm_install_plugins.md)
+- [list](../../../docs/reference/ocm_get_plugins.md)
+- [examine](../../../docs/reference/ocm_describe_plugins.md)
+- [remove](../../../docs/reference/ocm_install_plugins.md)
+
+plugins.
+
+Plugins can either be installed manually, just by copying the plugin executable to the plugin directory, or by using the CLI commands. They use OCM component versions as installation source. Plugins must have the artifact type `ocmPlugin` and follow the rules for providing multi-platform executables by using separate
```

Comment:
To me it does not make sense that plugins can only be installed as as a component version as an installation source, shouldnt the source be able to come from anything?

If I read this it sounds like if I want to use the cli install command and I have a file on the FS my options are

1. Dont use the CLI and move it manually 
2. Create a ComponentVersion, add the `ocmPlugin` resource and then reference that during the install command. That to me sounds very weird

Another naming thing: wouldnt the plugin have a "resource type" instead of an "artifact type"? maybe im confusing things here though